components/script/dom/customelementregistry.rs

```diff
@@ -916,7 +917,10 @@ fn run_upgrade_constructor(
     }
     rooted!(in(*cx) let mut construct_result = ptr::null_mut::<JSObject>());
     {
-        // Step 8.1 TODO when shadow DOM exists
+        // Step 8.1
```

Comment:
```suggestion
        // Step 8.1. If definition's disable shadow is true and element's shadow root is non-null, then throw a "NotSupportedError" DOMException.
```

Reply:
We've recently moved to including specification text in comments rather than just step numbers.